pkg/httpclient/client_test.go

```diff
@@ -0,0 +1,198 @@
+	require.NoError(t, err)
+
+	respBody, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Equal(t, body, string(respBody))
+}
+
+func TestPost(t *testing.T) {
+	type body_struct struct {
+		Cluster string
+	}
+	body := `{"cluster":"DC/OS"}`
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Errorf("Expected `POST` request, got `%s`", r.Method)
+		}
+		if r.URL.EscapedPath() != "/post" {
+			t.Errorf("expected request to `/post, got `%s`", r.URL.EscapedPath())
+		}
+
+		decoder := json.NewDecoder(r.Body)
+		var body_sent body_struct
+		err := decoder.Decode(&body_sent)
+		require.NoError(t, err)
+		defer r.Body.Close()
+
+		require.Equal(t, body_sent.Cluster, "DC/OS")
+	}))
+	defer ts.Close()
+
+	conf := config.New()
+	conf.SetURL(ts.URL)
+
+	client := New(conf)
+
+	resp, err := client.Post("/post", "application/json", strings.NewReader(body))
+	require.NoError(t, err)
+	require.Equal(t, resp.StatusCode, http.StatusOK)
+}
+
+func TestACSToken(t *testing.T) {
+	body := `{"cluster":"DC/OS"}`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			t.Errorf("Expected `GET` request, got `%s`", r.Method)
+		}
+		if r.Header.Get("Authorization") != "token=secret" {
+			t.Errorf("expected Authorization header to be `secret`, got `%s`", r.URL.EscapedPath())
+		}
+		if r.URL.EscapedPath() != "/get" {
+			t.Errorf("expected request to `/get, got `%s`", r.URL.EscapedPath())
+		}
+
+		w.Write([]byte(body))
+	}))
+	defer ts.Close()
+
+	conf := config.New()
+	conf.SetURL(ts.URL)
+	conf.SetACSToken("secret")
+
+	client := New(conf)
+
+	_, err := client.Get("/get")
+	require.NoError(t, err)
+}
+
+func TestNewRequestGet(t *testing.T) {
+	body := `{"cluster":"DC/OS"}`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			t.Errorf("Expected `GET` request, got `%s`", r.Method)
+		}
+		if r.URL.EscapedPath() != "/get" {
+			t.Errorf("expected request to `/get, got `%s`", r.URL.EscapedPath())
+		}
+
+		w.Write([]byte(body))
+	}))
+	defer ts.Close()
+
+	conf := config.New()
+	conf.SetURL(ts.URL)
+
+	client := New(conf)
+
+	req, err := client.NewRequest("GET", "/get", nil)
+	require.NoError(t, err)
+
+	resp, err := client.Do(req)
+	require.NoError(t, err)
+	respBody, err := ioutil.ReadAll(resp.Body)
+	require.Equal(t, body, string(respBody))
+}
+
+func TestNewRequestPost(t *testing.T) {
+	type body_struct struct {
+		Cluster string
+	}
+	body := `{"cluster":"DC/OS"}`
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Errorf("Expected `POST` request, got `%s`", r.Method)
+		}
+		if r.URL.EscapedPath() != "/post" {
+			t.Errorf("expected request to `/post, got `%s`", r.URL.EscapedPath())
+		}
+
+		decoder := json.NewDecoder(r.Body)
+		var body_sent body_struct
```

Comment:
I forgot to add linting in the CI, it complains here as underscores vars shouldn't be used. There are also 3 other locations in this file.